### PR TITLE
vm_xml.py: Add class to handle vcpus xml elements

### PR DESCRIPTION
**Pull request by Grace-chhu**
Add the class:VMVCPUsXML to handle vcpus xml elements:
  <vcpus>
    <vcpu id='0' enabled='yes' hotpluggable='no' order='1'/>
    <vcpu id='1' enabled='yes' hotpluggable='yes' order='2'/>
    <vcpu id='2' enabled='no' hotpluggable='yes' order='3'/>
  </vcpus>

Signed-off-by: Chenli Hu chhu@redhat.com

### CODE REVIEW

File: virttest/libvirt_xml/vm_xml.py [hunk 1]
```diff
@@ -2051,6 +2062,49 @@ def remove_feature(self, name):
...
+class VMVCPUsXML(base.LibvirtXMLBase):
+
+    """
+    VCPUs tag XML class
+
+    Elements:
+        vcpus: list of dict - id, enabled, hotpluggable, order
+    """
+
+    __slots__ = ('vcpus')
```
Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: ```__slots__ = ('vcpu',)```

File: virttest/libvirt_xml/vm_xml.py [hunk 2]
```diff
@@ -2051,6 +2062,49 @@ def remove_feature(self, name):
...
+
+    def __init__(self, virsh_instance=base.virsh):
+        accessors.XMLElementList('vcpus', self, parent_xpath="/",
```
Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: Inside the <vcpus>, this should be 'vcpu'

File: virttest/libvirt_xml/vm_xml.py (hunk 1 above)
Comment (Grace-chhu) [inline virttest/libvirt_xml/vm_xml.py]: Changed to:  __slots__ = ('vcpu')

File: virttest/libvirt_xml/vm_xml.py (hunk 2 above)
Comment (Grace-chhu) [inline virttest/libvirt_xml/vm_xml.py]: Changed to:  accessors.XMLElementList('vcpu', self, parent_xpath="/",

File: virttest/libvirt_xml/vm_xml.py [hunk 3]
```diff
@@ -2051,6 +2062,49 @@ def remove_feature(self, name):
             root.remove(remove_feature)
 
 
+class VMVCPUsXML(base.LibvirtXMLBase):
+
+    """
+    VCPUs tag XML class
+
+    Elements:
+        vcpus: list of dict - id, enabled, hotpluggable, order
```
Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: Also here

File: virttest/libvirt_xml/vm_xml.py [hunk 4]
```diff
@@ -2051,6 +2062,49 @@ def remove_feature(self, name):
...
+    """
+
+    __slots__ = ('vcpu')
+
+    def __init__(self, virsh_instance=base.virsh):
+        accessors.XMLElementList('vcpu', self, parent_xpath="/",
+                                 marshal_from=self.marshal_from_vcpus,
+                                 marshal_to=self.marshal_to_vcpus)
```
Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: And maybe you can also update these two functions new

File: virttest/libvirt_xml/vm_xml.py (hunk 3 above)
Comment (Grace-chhu) [inline virttest/libvirt_xml/vm_xml.py]: Modified to:
+    vcpus tag XML class
+
+    Elements:
+        vcpu: list of dict - id, enabled, hotpluggable, order

File: virttest/libvirt_xml/vm_xml.py (hunk 4 above)
Comment (Grace-chhu) [inline virttest/libvirt_xml/vm_xml.py]: Modified the two functions.

**Pull request merged**